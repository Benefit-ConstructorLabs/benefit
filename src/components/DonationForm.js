import React from 'react';
import PropTypes from 'prop-types';
import '../../styles/components/donation-form.scss';

const DonationForm = ({
  donationAmount,
  setDonationAmount,
  firstName,
  photo,
  togglePaymentDetails,
}) => (
  <React.Fragment>
    <img className="donation__recipient__photo" alt={firstName} src={photo} />
    <h3 className="donation__purpose">
      {`Donate to ${firstName}`}
    </h3>
    <form
      onSubmit={(event) => {
        event.preventDefault();
        togglePaymentDetails();
      }}
      className="donation__form"
    >
      <h3 className="donation__form__heading">How much?</h3>
      <p>
        <input
          type="text"
          className="nolabel donation__form__amount"
          id="donation__amount"
          value={donationAmount === undefined ? '£' : `£${donationAmount}`}
          onChange={event => setDonationAmount(event.target.value)}
        />
      </p>
      <p>
        <input onChange={event => setDonationAmount(event.target.value)} value={donationAmount} className="donation__form__slider" type="range" min="1" max="10" />
      </p>
      <p className="donation__form__quick-amount">
        <button onClick={event => setDonationAmount(event.target.value)} type="button" className="btn btn__quick-amount" value="1">£1</button>
        <button onClick={event => setDonationAmount(event.target.value)} type="button" className="btn btn__quick-amount" value="2">£2</button>
        <button onClick={event => setDonationAmount(event.target.value)} type="button" className="btn btn__quick-amount" value="5">£5</button>
      </p>
      <button className="btn btn__primary btn__submit" type="submit">
        Donate to
        {' '}
        {firstName}
      </button>
    </form>
  </React.Fragment>
);

DonationForm.propTypes = {
  donationAmount: PropTypes.number,
  setDonationAmount: PropTypes.func.isRequired,
  firstName: PropTypes.string.isRequired,
  photo: PropTypes.string.isRequired,
  togglePaymentDetails: PropTypes.func.isRequired,
};

DonationForm.defaultProps = {
  donationAmount: undefined,
};

export default DonationForm;
