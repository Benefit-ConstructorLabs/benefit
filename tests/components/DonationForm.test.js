import React from 'react'
import DonationForm from '../../src/components/DonationForm';
import renderer from 'react-test-renderer';

describe('App', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<DonationForm />)
      .toJSON()
    expect(tree).toMatchSnapshot();
  });
})