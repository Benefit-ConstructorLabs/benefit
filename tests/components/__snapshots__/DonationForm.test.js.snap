// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonationForm should render correctly 1`] = `
<div
  className="donation"
>
  <img
    alt="Recipient"
    className="donation__recipient__photo"
    src="https://via.placeholder.com/150?text=Recipient+photo"
  />
  <h3
    className="donation__purpose"
  >
    Donate towards a night in a shelter
  </h3>
  <form
    className="donation__form"
    onSubmit={[Function]}
  >
    <h4>
      How much?
    </h4>
    <p>
      <input
        className="donation__form__amount"
        id="donation__amount"
        placeholder="£1"
        type="number"
      />
    </p>
    <p>
      <input
        type="range"
      />
    </p>
    <p
      className="donation__form__quick-amount"
    >
      <button
        className="btn btn__quick-amount"
        type="button"
      >
        50p
      </button>
      <button
        className="btn btn__quick-amount"
        type="button"
      >
        £1
      </button>
      <button
        className="btn btn__quick-amount"
        type="button"
      >
        £2
      </button>
    </p>
    <button
      className="btn btn__submit"
      type="submit"
    >
      Donate to firstname
    </button>
  </form>
</div>
`;
